ethexe/runtime/common/src/journal.rs

```diff
@@ -324,7 +327,10 @@ impl<S: Storage> JournalHandler for NativeJournalHandler<'_, S> {
     }
 
     fn stop_processing(&mut self, _dispatch: StoredDispatch, _gas_burned: u64) {
-        todo!()
+        // This means we are out of gas for block, not for chunk.
+        if self.gas_allowance_counter.left() < self.chunk_gas_limit {
```

Comment:
this will not allow you to run next chunk with remaining_allowance.min(chunk_gas_limit), but you should. 
stop processing here are only notifying about some chunk passed across chunk allowance, isn't it?

Reply:
Hitting this condition `self.gas_allowance_counter.left() < self.chunk_gas_limit` means the current chunk started with a remaining gas allowance smaller than the chunk-gas limit (`allowance_counter` is updated only after the chunk processing finishes).

Because such chunk is running on the leftover gas budget,  this condition plus `stop_processing` journal note means we run out of gas for block.

![image](https://github.com/user-attachments/assets/9e96e101-6d07-4677-b237-aad79d0a56e7)